### PR TITLE
Feodo tracker#1103

### LINKED ISSUE #1103: [Analyzer] Feodo Tracker + allow to disable the analyzer/updates
We need to find a way (very similar to how TorProject analyzer is handled), to periodically download the dataset from the Feodo tracker and keep them inside IntelOwl to allow users to perform lookups inside it.

They provide 3 different datasets so we could choose to download all of them and show the results differently based on which list we found the looked-up IP address in.

Plus, considering that the download of this data is something heavy in terms of traffic generated, we should provide a variable (that can be changed) where the users can decide whether to enable it or not. In this way, users who are not interested in the service can disable the recurrent updates. As a result of this, we should show to the GUI the analyzer as "not configured" like the analyzers which don't have the API key set.
Then, the same behavior should be replicated to the other analyzers working like that (Tor, Maxmind, Yara_Scan_*, Talos).

### PR DESCRIPTION
**Pull request by g4ze**
closes #1103 
# Description

Added Feodo Tracker 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [x] I have read and understood the rules about [how to Contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) to this project
- [x] The pull request is for the branch `develop`
- [x] A new plugin (analyzer, connector, visualizer, playbook, pivot or ingestor) was added or changed, in which case:
    - [x] I strictly followed the documentation ["How to create a Plugin"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-add-a-new-plugin)
    - [x] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] [Advanced-Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Advanced-Usage.md) was updated (in case the plugin provides additional optional configuration).
    - [x] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [x] I have dumped the configuration from Django Admin using the `dumpplugin` command and added it in the project as a data migration. (["How to share a plugin with the community"](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-share-your-plugin-with-the-community))
    - [x] If a File analyzer was added and it supports a mimetype which is not already supported, you added a sample of that type inside the archive `test_files.zip` and you added the default tests for that mimetype in [test_classes.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_classes.py).
    - [x] If you created a new analyzer and it is free (does not require API keys), please add it in the `FREE_TO_USE_ANALYZERS` playbook by following [this guide](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-modify-a-plugin).
    - [x] Check if it could make sense to add that analyzer/connector to other [freely available playbooks](https://intelowl.readthedocs.io/en/develop/Usage.html#list-of-pre-built-playbooks).
    - [x] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results.
- [x] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowl.readthedocs.io/en/latest/Contribute.html#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [x] I have added tests for the feature/bug I solved (see `tests` folder). All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Contribute.md)).
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.

### Important Rules
- If you miss to compile the Checklist properly, your PR won't be reviewed by the maintainers.
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review. After being reviewed and received a "change request", you should explicitly ask for a review again once you have made the requested changes.
image
{"found":"false"}

### CODE REVIEW

**Comment by g4ze** [comment]
Hi there! @mlodic would a logic like work? Here, we would only update if `update_on_run` is true (false by default) or it has been more than 24hrs since last update or if the db simply has never been initiated.

**Review by mlodic — CHANGES_REQUESTED** [review]

File: api_app/analyzers_manager/observable_analyzers/feodo_tracker.py
Comment (g4ze) [reply api_app/analyzers_manager/observable_analyzers/feodo_tracker.py:64]: Apologies for the ignorance and a useless pr #2127 ; I will put more efforts in understanding and exploring the code base. Working on the periodic tasks.

Comment (g4ze) [reply api_app/analyzers_manager/observable_analyzers/feodo_tracker.py:64]: I think we also need to check if the db exists; even if the `update on run` is `false` a db should be initialised if it already does not exist.

Comment (mlodic) [reply api_app/analyzers_manager/observable_analyzers/feodo_tracker.py:64]: yep that makes sense

**Comment by g4ze** [comment]
hey! @mlodic I'm facing some frontend build issues, can you please check [this](https://honeynetpublic.slack.com/archives/C01KVGMAKL6/p1707897410925139).

**Review by mlodic — CHANGES_REQUESTED** [review]
great we are almost ready!

File: tests/test_crons.py
Comment (g4ze) [reply tests/test_crons.py:112]: oopsies, overlooked that; correcting.

**Comment by g4ze** [comment]
made required chages; is this test error arrising because we run `docker exec intelowl_uwsgi coverage run manage.py test --keepdb tests` with keepdb? The error is not arising on my local ci test.

File: api_app/analyzers_manager/migrations/0066_analyzer_config_feodo_tracker.py
Comment (mlodic) [inline api_app/analyzers_manager/migrations/0066_analyzer_config_feodo_tracker.py:61]: here too

File: api_app/analyzers_manager/migrations/0066_analyzer_config_feodo_tracker.py
Comment (mlodic) [inline api_app/analyzers_manager/migrations/0066_analyzer_config_feodo_tracker.py:72]: here too

**Review by mlodic — COMMENTED** [review]
I think we are good, except from the migration problem.

**Comment by g4ze** [comment]
works now, nice

File: api_app/analyzers_manager/migrations/0067_analyzer_config_feodo_tracker.py
Comment (mlodic) [inline api_app/analyzers_manager/migrations/0067_analyzer_config_feodo_tracker.py:28]: @0ssigeno is there a way to avoid to have hardcoded this pk, leading to possible conflicts?

**Comment by mlodic** [comment]
just waiting for @0ssigeno feedback then we can merge this. Ty!

**Comment by g4ze** [comment]
@mlodic in the meantime, can you suggest any other issue for me to take up?

**Comment by mlodic** [comment]
this is a simple task to change an already existing analyzer: https://github.com/intelowlproject/IntelOwl/issues/1955

this would require to change a bit the flow of the classic development of a new one so that could be a good little challenge

File: api_app/analyzers_manager/migrations/0067_analyzer_config_feodo_tracker.py
Comment (0ssigeno) [reply api_app/analyzers_manager/migrations/0067_analyzer_config_feodo_tracker.py:28]: Ok, this requires a change in the backend that changes the `python_module_pk` to something that is invariant on the number of python modules that you have in your db.

@g4ze, the pr is good, I will merge this once i have made some changes in the backend to correctly support the update function

**Review by 0ssigeno — CHANGES_REQUESTED** [review]

**Comment by 0ssigeno** [comment]
I should have fixed the issue

**Pull request merged**
